data_structures/is_even.py

```diff
@@ -0,0 +1,23 @@
+# Find if a number is even or odd,
+# without using the mod, div or multiplication operators.
+
+
+def is_even(n):
```

Comment:
No, this implementation is way, way too slow. You can do this in O(n).

Look into `&` bit operator and think about what bit to check to see if a number is even.

Reply:
There are three ways to optimize this code, I have tried the Bit wise operation AND, OR and XOR. All three were successful.